gips/data/prism/prism.py

```diff
@@ -262,27 +262,25 @@ def get_bil_vsifile(d, a):
                 imgs = []
                 asset_fns = [] # have to grab filenames for multiple days
                 for tileobj in inv.data.values():
-                    datobj = tileobj.tiles.values()[0]
-                    asset_fns.append(
-                        os.path.basename(datobj.assets['_ppt'].filename))
-                    imgs.append(GeoImage(get_bil_vsifile(datobj, '_ppt')))
+                    data_obj = next(iter(tileobj.tiles.values()))
+                    asset_fns.append(os.path.basename(data_obj.assets['_ppt'].filename))
+                    imgs.append(GeoImage(get_bil_vsifile(data_obj, '_ppt')))
 
                 with self.make_temp_proc_dir() as tmp_dir:
                     tmp_fp = os.path.join(tmp_dir, prod_fn)
-                    oimg = GeoImage(tmp_fp, imgs[0])
-                    oimg.SetNoData(-9999)
-                    oimg.SetBandName(
+                    oimg = GeoImage.create_from(imgs[0], tmp_fp)
+                    oimg.set_nodata(-9999)
+                    oimg.set_bandname(
                         description + '({} day window)'.format(lag), 1
                     )
-                    oimg.SetMeta(self.prep_meta(sorted(asset_fns)))
-                    for chunk in oimg.Chunks():
-                        oarr = oimg[0].Read(chunk) * 0.0 # wat
+                    oimg.add_meta(self.prep_meta(sorted(asset_fns)))
+                    for chunk in oimg.chunks():
+                        oarr = oimg[0].read(chunk) * 0.0 # wat
```

Comment:
`wat` is right.  This is probably a bug, and why pptsum ends up with some pixels having large negative values, by adding no-data values (-9999) together.  Not that this holds this up.

Reply:
also mark `pptsum` deprecated

Reply:
internal issue filed